### ISSUE TITLE
Use a lot of disk space

### ISSUE THREAD

**Issue by sn00pyd0g3** [github/codeql-go#551]
When autobuilder compiles the Go source code, it will pull the package code and generate a binary file. This consumes a lot of disk space. Can these files be deleted automatically after the scan is completed?

**Comment by hmakholm** [comment]
Yes, it is safe to delete built files as soon as information for the CodeQL database has been collected.

If you're using Actions, you can add a step to your workflow file with appropriate commands to clean the source tree after your build steps but before the `codeql-action/analyze` action.

If you're using the CodeQL CLI directly, you can clean up built artifacts immediately after the `codeql database create` or `codeql database trace-command` invocation.

The autobuilders don't to this automatically, out of concern that deleting _too much_ might break things in use cases where CodeQL is only one part of a larger CI or build constellation that might need the built files later. It generally feels safer to leave the decision about what to delete to someone who knows the actual project being analyzed.
